src/components/AudioPlayer/AudioPlayer.js

```diff
@@ -171,9 +187,14 @@ class AudioPlayerSub extends Component {
     if (active && !nextProps.active) {
       // pause track if needed
       if (playing && !nextProps.keepPlaying) {
-        TrackPlayer.pause()
-        updatePlaying(false)
-        this.setState({ startSwitch: true })
+        this.setState({ startSwitch: true }, () => {
+          TrackPlayer.pause().then(() => {
+            updatePlaying(false)
+          }).catch(err => {
+            console.error(err)
+          })
+        })
```

Comment:
I haven't seen this component before so I don't have all the context, but I don't think I've seen a `setState` with a callback  as as second parameter before. That'll run after the state is updated? or how's it supposed to work? 

I get that the track pause is async, but I'd think that what we'd want would be 
```
this.setState({ startSwitch : true })
TrackPlayer.pause().then(()=>updatePlaying(false)).catch(err=>console.error(err))

```

Reply:
Correct, you can pass a callback to setState.  The reason for the code change here was to get a sequence of events.  Do you think otherwise?

Reply:
Nah, that's good, I needed to re-read some of react docs to remember there was that second param callback. 
This is good.